toxcore/tox_options.h

```diff
@@ -310,6 +350,18 @@ struct Tox_Options {
      * @private
      */
     char *owned_proxy_host;
+
+    /**
+     * @brief Owned pointer to the SOCKS5 proxy username.
+     * @private
+     */
+    uint8_t *owned_proxy_socks5_username;
```

Comment:
We're introducing new data members here. Do we need them to not be owned by default as well?

Reply:
That would make the API inconsistent - some things are not owned and require a flag to be set, but others are owned.

Reply:
Does it functionally matter? If clients think it's not owned, they will save their pointers. If they enable the flag *or* happen to know that it is owned, they don't. It's making things more complicated, and 0.3.0 will get rid of this un-owned mode entirely.

Reply:
I believe I got this addressed now.

Per the IRC discussion:

```
<iphy> Maybe name the fields internal_do_not_set_directly_proxy_socks5_username etc
<iphy> No new code should be setting those fields, and we shouldn't build complexity to cover for that case
```